spring-cloud-task-core/src/test/java/org/springframework/cloud/task/repository/dao/BaseTaskExecutionDaoTestCases.java

```diff
@@ -0,0 +1,305 @@
+		assertEquals(2, dateTime.get(Calendar.MONTH) + 1);
+		assertEquals(22, dateTime.get(Calendar.DAY_OF_MONTH));
+		assertEquals(23, dateTime.get(Calendar.HOUR_OF_DAY));
+		assertEquals(59, dateTime.get(Calendar.MINUTE));
+		assertEquals(0, dateTime.get(Calendar.SECOND));
+	}
+
+	@Test
+	@DirtiesContext
+	public void getLatestTaskExecutionsByTaskNamesWithMultipleTaskNames()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final List<TaskExecution> latestTaskExecutions = dao.getLatestTaskExecutionsByTaskNames("FOO1", "FOO3", "FOO4");
+		assertTrue("Expected only 1 taskExecution but got " + latestTaskExecutions.size(), latestTaskExecutions.size() == 3);
+
+		final Calendar dateTimeFoo3 = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+		dateTimeFoo3.setTime(latestTaskExecutions.get(0).getStartTime());
+
+		assertEquals(2016, dateTimeFoo3.get(Calendar.YEAR));
+		assertEquals(8, dateTimeFoo3.get(Calendar.MONTH) + 1);
+		assertEquals(20, dateTimeFoo3.get(Calendar.DAY_OF_MONTH));
+		assertEquals(14, dateTimeFoo3.get(Calendar.HOUR_OF_DAY));
+		assertEquals(45, dateTimeFoo3.get(Calendar.MINUTE));
+		assertEquals(0, dateTimeFoo3.get(Calendar.SECOND));
+
+		final Calendar dateTimeFoo1 = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+		dateTimeFoo1.setTime(latestTaskExecutions.get(1).getStartTime());
+
+		assertEquals(2015, dateTimeFoo1.get(Calendar.YEAR));
+		assertEquals(2, dateTimeFoo1.get(Calendar.MONTH) + 1);
+		assertEquals(22, dateTimeFoo1.get(Calendar.DAY_OF_MONTH));
+		assertEquals(23, dateTimeFoo1.get(Calendar.HOUR_OF_DAY));
+		assertEquals(59, dateTimeFoo1.get(Calendar.MINUTE));
+		assertEquals(0, dateTimeFoo1.get(Calendar.SECOND));
+
+		final Calendar dateTimeFoo4 = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+		dateTimeFoo4.setTime(latestTaskExecutions.get(2).getStartTime());
+
+		assertEquals(2015, dateTimeFoo4.get(Calendar.YEAR));
+		assertEquals(2, dateTimeFoo4.get(Calendar.MONTH) + 1);
+		assertEquals(20, dateTimeFoo4.get(Calendar.DAY_OF_MONTH));
+		assertEquals(14, dateTimeFoo4.get(Calendar.HOUR_OF_DAY));
+		assertEquals(45, dateTimeFoo4.get(Calendar.MINUTE));
+		assertEquals(0, dateTimeFoo4.get(Calendar.SECOND));
+	}
+
+	/**
+	 * This test is a special use-case. While not common, it is theoretically possible, that a task may have
+	 * executed with the exact same start time multiple times. In that case we should still only get 1 returned
+	 * {@link TaskExecution}.
+	 */
+	@Test
+	@DirtiesContext
+	public void getLatestTaskExecutionsByTaskNamesWithIdenticalTaskExecutions()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final List<TaskExecution> latestTaskExecutions = dao.getLatestTaskExecutionsByTaskNames("FOO5");
+		assertTrue("Expected only 1 taskExecution but got " + latestTaskExecutions.size(), latestTaskExecutions.size() == 1);
+
+		final Calendar dateTime = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+		dateTime.setTime(latestTaskExecutions.get(0).getStartTime());
```

Comment:
We may want to check the executionId to make sure it is the one we are expecting here.  Since it is the only differentiator from the other executions.